src/components/Select/types.ts

```diff
@@ -70,6 +74,17 @@ export type SelectProps<T = any> = QAProps &
             | React.ReactElement<SelectOptionGroup<T>, typeof OptionGroup>[];
     };
 
+export type SelectBasicProps<Option = any> = Omit<SelectProps<Option>, 'options'> & {
+    options?: SelectOptionType<Option>[];
+};
+
+export type SelectAsyncProps<Option = any, Pagination = any> = Omit<
+    SelectProps<Option>,
+    'options'
+> & {
+    options: SelectAsyncOptionType<SelectOptionType<Option>[], Pagination>;
```

Comment:
What about initial options?

Reply:
I don't think that this is good idea. It will just extends number of props and introduce ambiguities. You can return initialOptions in async function you pass to option.

We can highlight it in additional story.